src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -118,6 +119,36 @@ object Instance {
     def apply(runSpec: RunSpec): Instance = Scheduled(runSpec, Id.forRunSpec(runSpec.id))
   }
 
+  object Provisioned {
+    /**
+      * Factory method for creating provisioned instance from Scheduled instance for apps
+      * @param scheduledInstance instance in a Scheduled state
+      * @return new instance in a provisioned state
+      */
+    def apply(scheduledInstance: Instance, agentInfo: AgentInfo, networkInfo: core.task.state.NetworkInfo, app: AppDefinition)(implicit clock: Clock): Instance = {
+      require(scheduledInstance.isScheduled, "Scheduled instance is required to create provisioned instance.")
+
+      val task = Task(
+        taskId = Task.Id.forInstanceId(scheduledInstance.instanceId, None),
+        runSpecVersion = app.version,
+        status = Task.Status(
+          stagedAt = clock.now(),
+          condition = Condition.Created,
+          networkInfo = networkInfo
+        )
+      )
+
+      val tasksMap = Map(task.taskId -> task)
+
+      scheduledInstance.copy(
+        agentInfo = Some(agentInfo),
+        state = Instance.InstanceState(Condition.Provisioned, clock.now(), None, None),
+        tasksMap = tasksMap,
+        runSpecVersion = app.version // Note: The app version might change.
```

Comment:
The comment is wrong. `Provisioned` instances won't change their app version.

Reply:
oh ok, I copypasted this from another place in code :)